### DISCUSSION TITLE
React Query v4

### DISCUSSION THREAD

**Discussion by johnf** [pmndrs/jotai#1312] [Ideas] [answered]
This is more of an FYI.
As of v4 React Query has moved from ```react-query``` to ```@tanstack/react-query``` so import will need to change to move from v3 to v4

**Comment by dai-shi** [comment, marked as answer]
We have a PR #1126 filed a few months ago, which is finally merged recently.
It will be included in [v1.7.7](https://github.com/pmndrs/jotai/milestone/33).

You (or anyone looking at this) are highly encouraged to try the codesandbox build to see if it works for you now.
https://ci.codesandbox.io/status/pmndrs/jotai/pr/1126/builds/274516
☝️ Find "Local Install Instructions"

**Reply by ericberens** [reply]
Do you have any reference into your release process? I see that this is a breaking change but released as a patch? I'm unsure if you follow semver or other but would be helpful to understand how to manage upgrades.

**Reply by dai-shi** [reply]
We try best to follow semver for `jotai`.
For `jotai/*`, we can't really follow semver.
We should probably consider separating different libs in the future. (For example, moving `jotai/query` to [jotai-labs](https://github.com/jotai-labs), or maybe [tanstack](https://github.com/tanstack).)

**Reply by ericberens** [reply]
I had assume this would be quite difficult given the various supported peerDependencies but appreciate the reply and thoughts.
